text/0000-sbom.md

```diff
@@ -0,0 +1,125 @@
+
+`io.buildpacks.bom.buildpack.id` - Buildpack ID for the buildpack that created the BOM
+`io.buildpacks.bom.types` - Set to a comma seperated list of `build` or `launch` as applicable.
+`io.buildpacks.bom.layer.name` (Optional) - Set to the name of the layer if the `bom` was associated with a specific layer. 
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+This RFC adds a fair bit of complexity to the lifecycle and makes it responsible for more than just buildpack orchestration and container/layer assembly.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## Use a different SBOM format (SPDX)
+
+SPDX is the other leading SBOM format. It is a Linux Foundation project and is another front-runner in the SBOM eco-system. SPDX is heavily focused on compliance related metadata whereas CycloneDX is focused on security and supply chain analysis. SPDX also doesn't have as rich of an ecosystem when it comes to tooling for generating SBOMs as CycloneDX.
+
+For reference see - https://cyclonedx.org/tool-center/
+
+A more complete analysis of leading SPDX and CycloneDX tooling relevant to Buildpacks is documented at [sophiewigmore/bom](https://github.com/sophiewigmore/bom). This includes common SBOM field mapping, example SBOM outputs, scanning times.
+
+## Delegate SBOM merging to platform
+
+The main reason that the lifecycle needs to have a hard dependency on CycloneDX in this proposal is because we propose that the lifecycle merges the SBOM from various buildpacks and layers. If the lifecycle were to not do this, the content of the SBOM can be completely opaque to the lifecycle. This might provide greater flexibility and almost no dependency on the lifecycle side. Instead a platform like `pack` can be responsible for gathering the individual SBOM files and merging them together for the user. This can also be provided as a reusable library/CLI so that other platforms could do the same.
+
+
+# Prior Art
+[prior-art]: #prior-art
+
+- [#RFC 53](https://github.com/buildpacks/rfcs/blob/main/text/0053-decouple-buildpack-plan-and-bom.md)
+- [#RFC 87](https://github.com/buildpacks/rfcs/blob/main/text/0087-bom-in-layer-metadata.md)
+
+# Unresolved Questions
+[unresolved-questions]: #unresolved-questions
+
+- Implementation details about how the lifecycle should be merging the SBOMs.
+- Moving the SBOM from the `io.buildpacks.build.metadata` label to a file in the output image to avoid large labels
```

Comment:
Is there a way to ensure that the output file (with the BOM) and the image are always linked together? Would we put the SHA of the BOM file in a label on the image?

Reply:
The current RFC just puts the combined BOM in a label as usual. I was hoping to have a separate RFC that moves it to a file which would do something similar (i.e. put the SHA of the BOM file in a label on the image)